torch/lib/THD/base/data_channels/GlooCache.hpp

```diff
@@ -375,31 +391,32 @@ struct algorithm_spec<CollectiveType::BROADCAST, T> {
 #ifdef WITH_CUDA
     } else if (device == DeviceType::CUDA) {
       auto stream = THCState_getCurrentStream(THDGetCudaState());
-      
-      #if defined(GLOO_USE_IBVERBS) && GLOO_USE_IBVERBS
-            // Only enable GPU direct if the device supports it
-            if (context->getDevice()->hasGPUDirect()) {
-              algo = std::make_shared<::gloo::CudaBroadcastOneToAll<T,
-                                      ::gloo::CudaDeviceWorkspace<T>>>(
-                context,
-                std::initializer_list<T*>{reinterpret_cast<T*>(input_buffer.get())},
-                count,
-                src_rank,
-                0,
-                std::vector<cudaStream_t>{stream});
-            } else
-      #endif
-            {
-              algo = std::make_shared<::gloo::CudaBroadcastOneToAll<T,
-                                      ::gloo::CudaHostWorkspace<T>>>( 
-                context,
-                std::initializer_list<T*>{reinterpret_cast<T*>(input_buffer.get())},
-                count,
-                src_rank,
-                0,
-                std::vector<cudaStream_t>{stream});
-            }
+
+#if defined(WITH_GLOO_IBVERBS) && WITH_GLOO_IBVERBS
+      // Only enable GPU direct if the device supports it
+      if (context->getDevice()->hasGPUDirect()) {
+        algo = std::make_shared<::gloo::CudaBroadcastOneToAll<T,
+                                ::gloo::CudaDeviceWorkspace<T>>>(
+          context,
+          std::initializer_list<T*>{reinterpret_cast<T*>(input_buffer.get())},
+          count,
+          src_rank,
+          0,
+          std::vector<cudaStream_t>{stream});
+      } else
 #endif
+      {
+        algo = std::make_shared<::gloo::CudaBroadcastOneToAll<T,
+                                ::gloo::CudaHostWorkspace<T>>>(
+          context,
+          std::initializer_list<T*>{reinterpret_cast<T*>(input_buffer.get())},
+          count,
+          src_rank,
+          0,
+          std::vector<cudaStream_t>{stream});
+      }
+#endif
```

Comment:
an unnecessary endif or does it match WITH_CUDA?

Reply:
It matches with WITH_CUDA.